### PR TITLE
Feature/watch later

### PR DESCRIPTION
**Pull request by Kevin-Solomon**
# ❓What is the change ❓
- Add a menu component where users can add to watch later in the video listing page.
- Users can also add to watch later on the single video page. 
- Users can not add the same video to the list.
# What is it ❓
 - [ ] bug🐛
 - [x] feature 👩‍💻
 - [ ] fix 🔧
 ## To access these features user must be logged in

### CODE REVIEW

File: src/components/LandscapeVideoCard/LandscapeVideoCard.jsx [hunk 1]
```diff
@@ -22,17 +23,22 @@ function LandscapeVideoCard({
...
 }) {
   const { authState } = useAuth();
   const { historyDispatch } = useHistory();
   const { likeDispatch } = useLike();
+  const { watchLaterDispatch } = useWatchLater();
   const deleteFromList = () => {
     if (!!inHistory) {
       deleteFromHistory(_id, authState.token, historyDispatch);
     }
     if (!!inLikedVideo) {
```
Comment (VishalPatil18) [inline src/components/LandscapeVideoCard/LandscapeVideoCard.jsx:36]: Why `!!`? Like what is it's usecase here?

File: src/components/VideoCard/VideoCard.jsx [hunk 2]
```diff
@@ -77,7 +98,33 @@ function VideoCard({
...
+            <span
+              onClick={e => {
+                e.stopPropagation();
+                setDisplay(prevState => !prevState);
+              }}
+            >
+              {getIcons('VIDEO_MENU', '20px')}
+            </span>
+            <div
+              className={display ? 'menu-options show-menu' : 'menu-options '}
```
Comment (VishalPatil18) [inline src/components/VideoCard/VideoCard.jsx:111]: as `menu-options` is the class which is common you could write this as
```jsx
className={`menu-options ${display ? "show-menu" : ""}`}
```

This follows the DRY(do not repeat yourself) method

**Review by mohammed-sas — APPROVED** [review]
watchlater reducer and its utils has been coded well
functionalities are working fine
code is clean

**Review by VishalPatil18 — APPROVED** [review]
- Without login, if I try to add a video to watch later it shows that the video already exists in Watch later which I think is a bug and needs to be resolved soon.
- If I add a video to watch later from a single video page the alert doesn't appear on that page but comes when the user goes to the home page.

Just a few suggestions added. The rest looks quite good and works well. Great work @Kevin-Solomon

File: src/components/LandscapeVideoCard/LandscapeVideoCard.jsx (hunk 1 above)
Comment (Kevin-Solomon) [reply src/components/LandscapeVideoCard/LandscapeVideoCard.jsx:36]: It is redundant as it naturally coerces to boolean.

**Pull request merged**